framework/src/application.ts

```diff
@@ -156,27 +162,34 @@ export class Application {
 	}
 
 	public registerPlugin(
-		pluginKlass: typeof BasePlugin,
-		options: PluginOptions = {
-			loadAsChildProcess: false,
-		},
-		alias?: string,
+		pluginKlass: InstantiablePlugin<BasePlugin>,
+		options: PluginOptions = { loadAsChildProcess: false },
 	): void {
 		assert(pluginKlass, 'Plugin implementation is required');
 		assert(typeof options === 'object', 'Plugin options must be provided or set to empty object.');
-		assert(alias ?? pluginKlass.alias, 'Plugin alias must be provided.');
-		const pluginAlias = alias ?? pluginKlass.alias;
+		validatePluginSpec(pluginKlass);
+
+		const pluginAlias = options?.alias ?? pluginKlass.alias;
```

Comment:
why `alias` needs to be overwritten?

Reply:
It was already there to register a plugin with different alias. Say you imported a plugin and want to use it with own prefix alias to avoid any future conflict.